cli/command/context/create_test.go

```diff
@@ -0,0 +1,230 @@
+package context
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/context/docker"
+	"github.com/docker/cli/cli/context/kubernetes"
+	"github.com/docker/cli/cli/context/store"
+	"gotest.tools/assert"
+	"gotest.tools/env"
+)
+
+type cliMock struct {
+	command.Cli
+	contextStore   store.Store
+	currentContext string
+	configFile     *configfile.ConfigFile
+	in             func() *command.InStream
+	out            func() *command.OutStream
+}
+
+func (m *cliMock) ContextStore() store.Store {
+	return m.contextStore
+}
+
+func (m *cliMock) CurrentContext() string {
+	return m.currentContext
+}
+
+func (m *cliMock) ConfigFile() *configfile.ConfigFile {
+	return m.configFile
+}
+
+func (m *cliMock) In() *command.InStream {
+	return m.in()
+}
+
+func (m *cliMock) Out() *command.OutStream {
+	return m.out()
+}
+
+func makeCliMock(t *testing.T, opts ...func(*cliMock)) (*cliMock, func(), error) {
+	dir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
```

Comment:
Why taking `t` as an argument and return this err? You can assert directly on this err no? 🤔